### PR TITLE
NWPS-1018-bugfix/nwps-1018-Production-Issue-Logging

### PR DESCRIPTION
**Pull request by sankar-shunmuga**
please review it

### CODE REVIEW

File: repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/cta-card.ftl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <#macro ctaCard ctaCard>
     <#if ctaCard??>
         <#assign openInNewWindow=false/>
-        <#if ctaCard.ctaCardContentBlock.ctaLink.document??>
+        <#if ctaCard.ctaCardContentBlock?? && ctaCard.ctaCardContentBlock.ctaLink.document??>
```
Comment (smanivasag) [inline repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/cta-card.ftl:6]: Including the condition `ctaCard.ctaCardContentBlock??` on `<if>` makes it to move the control to `<else>` block and fails on `<#assign link = "${ctaCard.ctaCardContentBlock.ctaLink.url}">` (line number `12`)
```
[WARNING] [talledLocalContainer] %d{HH:mm:ss} WARN  [HstFreemarkerServlet] The following has evaluated to null or missing:
[INFO] [talledLocalContainer] ==> ctaCard.ctaCardContentBlock  [in template "webfile:/freemarker/hee/macros/cta-card.ftl" at line 12, column 36]
```
Could we instead wrap the whole if/else block with the following:

```
<#if ctaCard.ctaCardContentBlock??>
   ...
   ...
</#if>
```
Thanks!

File: repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/image-with-caption.ftl [hunk 2]
```diff
@@ -3,11 +3,13 @@
 <#macro imageWithCaption imageWithCaption>
     <@hst.link var="imgLink" hippobean=imageWithCaption/>
     <figure class="nhsuk-image" style="width: 100%">
-        <img class="nhsuk-image__img" src="${imgLink}" alt="${imageWithCaption.description}">
-        <#if imageWithCaption.caption?has_content>
-            <figcaption class="nhsuk-image__caption">
-                ${imageWithCaption.caption}
-            </figcaption>
+        <#if imageWithCaption??>
```
Comment (smanivasag) [inline repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/image-with-caption.ftl:6]: Seems like `[INFO] [talledLocalContainer] ==> imageWithCaption.description  [in template "webfile:/freemarker/hee/macros/image-with-caption.ftl" at line 7, column 67]` error has been caused by non-existence of `hippogallery:description` property on `hee:imagesetWithCaption` node [which essentially happens when the Editor uploads an image, but doesn't add a description (as it is optional anyway)] and so I afraid wrapping the block `<#if imageWithCaption??>` may not work in this case.

I think rendering `imageWithCaption.description` with default-value operator (https://freemarker.apache.org/docs/dgui_template_exp.html#dgui_template_exp_missing_default) might help us here:
```
<img class="nhsuk-image__img" src="${imgLink}" alt="${imageWithCaption.description!}">
```
Could we try this ? Thanks!

**Comment by smanivasag** [comment]
Hi @sankar-shunmuga - Thanks for the PR, have left a couple of comments. Could you take a look ? thanks

File: repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/cta-card.ftl (hunk 1 above)
Comment (sankar-shunmuga) [reply repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/cta-card.ftl]: Thanks @smanivasag , updated as per requested

File: repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/image-with-caption.ftl (hunk 2 above)
Comment (sankar-shunmuga) [reply repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/image-with-caption.ftl]: Thanks @smanivasag , updated as per requested

**Review by smanivasag — APPROVED** [review]

**Pull request merged**